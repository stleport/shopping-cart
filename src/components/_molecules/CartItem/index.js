import React from 'react';
import PropTypes from 'prop-types';
import ProductPicture from '../../_atoms/ProductPicture';
import ProductDescription from '../ProductDescription';
import ProductButton from '../../_atoms/ProductButton';

const CartItem = ({
  cart,
  cartItem,
  onAddToCart,
  onSubQuantity
}) => {
  const item = cart.items.find(c => c.productId === cartItem.productId);
  return (
    <div className="st-Item">
      <div className="st-Item__image st-Item__image--small">
        <ProductPicture source={cartItem.images[0]} altText={cartItem.title} />
      </div>
      <div className="st-Item__content">
        <ProductDescription 
          title={cartItem.title} 
          description={cartItem.description} 
          quantity={item.quantity} 
        />
        <div className="st-Item__extra">
          <ProductButton 
            iconName="minus"
            iconAlign="right" 
            iconColor="white"
            bgColor="dark"
            onSubQuantity={
              () => onSubQuantity(cartItem.productId, item.quantity)
            }
          />
          <ProductButton 
            iconName="plus"
            iconAlign="right"
            iconColor="dark"
            bgColor="light"
            onAddToCart={() => onAddToCart(cartItem.productId)}
          />
        </div>
      </div>
    </div>
  );
};

CartItem.propTypes = {
  cartItem: PropTypes.shape({
    productId: PropTypes.number,
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    images: PropTypes.arrayOf(PropTypes.string)
  }).isRequired,
  cart: PropTypes.shape({
    items: PropTypes.arrayOf(PropTypes.shape({
      productId: PropTypes.number.isRequired,
      quantity: PropTypes.number.isRequired
    }))
  }).isRequired,
  onAddToCart: PropTypes.func.isRequired,
  onSubQuantity: PropTypes.func.isRequired
};

export default CartItem;
