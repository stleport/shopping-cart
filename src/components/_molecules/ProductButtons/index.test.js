import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import CartItem from '.';

function ProductButtons() {
  const props = {
    onAddToCart: jest.fn(),
    onSubQuantity: jest.fn(),
    onRemoveFromCart: jest.fn(),
    quantity: 2
  }; 

  const CartItemJSX = (
    <CartItem {...props} />
  );

  const CartItemComponent = shallow(CartItemJSX);
  return {
    CartItemComponent,
    CartItemJSX,
    props
  };
}

describe('CartItem', () => {
  it('renders correctly CartItem component', () => {
    const { CartItemJSX } = ProductButtons();
    const CartItemComponent = renderer.create(CartItemJSX).toJSON();
    expect(CartItemComponent).toMatchSnapshot(); 
  });

  it('should contain two buttons', () => {
    const { CartItemComponent } = ProductButtons();
    expect(CartItemComponent.find('ProductButton').length).toEqual(2);
  });
});
