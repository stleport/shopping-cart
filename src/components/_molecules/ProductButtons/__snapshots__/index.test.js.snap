// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem renders correctly CartItem component 1`] = `
<div
  className="st-Item__actions"
>
  <button
    className="st-Edit-button__button st-Edit-button__button--dark"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="minus"
      className="st-Edit-button__icon--white"
      fill="white"
      src="minus.svg"
    />
  </button>
  <button
    className="st-Edit-button__button st-Edit-button__button--light"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="plus"
      className="st-Edit-button__icon--dark"
      fill="white"
      src="plus.svg"
    />
  </button>
</div>
`;
